app/controllers/concerns/acceptable.rb

```diff
@@ -0,0 +1,24 @@
+module Acceptable
+  extend ActiveSupport::Concern
+
+  # Format constants
+  ACCEPTABLE_FORMATS = %i[json].freeze
+
+  included do
+    self.responder = ApplicationResponder
+    respond_to(*ACCEPTABLE_FORMATS)
+    before_action :validate_requested_format!, :validate_provided_format!
+  end
+
+  # Checks request format and defaults or returns HTTP[406].
+  def validate_requested_format!
+    return if ACCEPTABLE_FORMATS.include?(request.format.symbol)
```

Comment:
I'm not sure if `request.format` is always set. It would make sense, I just don't know what the actual behavior is. From [1], I would say that `Mime::NullType` is the fallback ... but according to [2] that does not respond to `symbol`.

[1] https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/http/mime_negotiation.rb#L58
[2] https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/http/mime_type.rb#L322